### PR TITLE
Upsert capability for update in ChunkStore

### PR DESCRIPTION
**Pull request by bmoscon**
fix for #209

### CODE REVIEW

**Comment by bmoscon** [comment]
@jamesblackburn @TomTaylorLondon

**Comment by bmoscon** [comment]
@Wilfred

File: arctic/chunkstore/chunkstore.py [hunk 1]
```diff
@@ -363,8 +363,15 @@ def update(self, symbol, item, chunk_range=None):
             range and overwrite it with the data in item. This allows the user
             to update with data that might only be a subset of the
             original data.
+        upsert: True or False
```
Comment (Wilfred) [inline arctic/chunkstore/chunkstore.py]: True or False: in other words, bool :)

File: arctic/chunkstore/chunkstore.py [hunk 2]
```diff
@@ -363,8 +363,15 @@ def update(self, symbol, item, chunk_range=None):
...
+            if True, will write the data even if the symbol does not exist.
+            If true and an upsert is performed, chunk_range will be ignored
+        chunk_size: ?
```
Comment (Wilfred) [inline arctic/chunkstore/chunkstore.py]: Presumably this is an integer?

File: arctic/chunkstore/chunkstore.py (hunk 1 above)
Comment (bmoscon) [inline arctic/chunkstore/chunkstore.py]: point taken, will change ;)

File: arctic/chunkstore/chunkstore.py [hunk 3]
```diff
@@ -346,7 +346,7 @@ def append(self, symbol, item):
         """
         self.__update(symbol, item, combine_method=self.__concat)
 
-    def update(self, symbol, item, chunk_range=None):
+    def update(self, symbol, item, chunk_range=None, upsert=False, chunk_size=None):
```
Comment (Wilfred) [inline arctic/chunkstore/chunkstore.py]: Super pedantic: I might be tempted to keep the chunking arguments together.

File: arctic/chunkstore/chunkstore.py (hunk 2 above)
Comment (bmoscon) [inline arctic/chunkstore/chunkstore.py]: it depends on the chunker. for the date chunker, its an arctic DateRange object, or a Pandas DatetimeIndex

**Comment by Wilfred** [comment]
Apart from my minor nits, LGTM :+1:

File: arctic/chunkstore/chunkstore.py (hunk 3 above)
Comment (bmoscon) [inline arctic/chunkstore/chunkstore.py]: I can see your point, to me it makes more sense like so, after all you only have to specify a chunksize if you override upsert and set it to True

**Pull request merged**